src/main/java/org/wahlzeit_revisited/repository/DataObject.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2006-2009 by Dirk Riehle, http://dirkriehle.com
+ *
+ * This file is part of the Wahlzeit photo rating application.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+package org.wahlzeit_revisited.repository;
+
+
+/**
+ * A simple abstract implementation of Persistent with write count and dirty bit.
+ * Also defines (but does not use) the field "ID" for subclass use.
+ */
+public abstract class DataObject implements Persistent {
```

Comment:
When I understood correctly, we will move away from caching the whole data objects in memory, right? I'd assume we can remove `writeCount` and all derived methods.

Reply:
Deleted in: f41efd892f9e1b9e2910cdd7be1b603e22ae0f2d

Reply:
Caching is not important to teaching so if you want it gone it probably is a good idea. Performance doesn't matter.